drivers/entropy/entropy_litex.c

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <device.h>
+#include <drivers/entropy.h>
+#include <errno.h>
+#include <init.h>
+#include <soc.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define SUBREG_SIZE_BIT 8
+
+struct rng_prbs_dev_cfg {
+	volatile u32_t *reg_status;
+	volatile u32_t reg_width;
+};
+
+#define DEV_CFG(dev) \
+	((const struct rng_prbs_dev_cfg *const)(dev)->config->config_info)
+
+static inline unsigned int prbs_read(volatile u32_t *reg_status,
+					 volatile u32_t reg_width)
+{
+	u32_t shifted_data, shift, i;
+	u32_t result = 0;
+
+	for (i = 0; i < reg_width; ++i) {
+		shifted_data = *(reg_status + i);
+		shift = (reg_width - i - 1) * SUBREG_SIZE_BIT;
+		result |= (shifted_data << shift);
+	}
+
+	return result;
+}
+
+static int entropy_prbs_get_entropy(struct device *dev, u8_t *buffer,
+					 u16_t length)
+{
+	const struct rng_prbs_dev_cfg *const prbs_cfg = DEV_CFG(dev);
+
+	while (length > 0) {
+		size_t to_copy;
+		u32_t value;
+
+		value = prbs_read(prbs_cfg->reg_status, prbs_cfg->reg_width);
+		to_copy = MIN(length, sizeof(value));
+
+		memcpy(buffer, &value, to_copy);
+		buffer += to_copy;
+		length -= to_copy;
+	}
+	return 0;
+}
+
+static int entropy_prbs_init(struct device *dev)
+{
+	return 0;
+}
+
+static const struct entropy_driver_api entropy_prbs_api = {
+	.get_entropy = entropy_prbs_get_entropy
+};
+
+static const struct rng_prbs_dev_cfg rng_prbs_cfg = {
+	.reg_status = (volatile uint32_t *)DT_INST_0_LITEX_PRBS_BASE_ADDRESS,
+	.reg_width = DT_INST_0_LITEX_PRBS_SIZE
+};
```

Comment:
Since there is going to be only one instance of the entropy device, why not just directly reference these values (preferably mapped to shorter local definitions) in the `entropy_prbs_get_entropy` function?

Reply:
Removed the `rng_prbs_cfg` struct.